### PR TITLE
Import external result

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
Code to test - 
```
from modelon.impact.client import Client

client = Client(url="http://127.0.0.1:8080/")
workspace = client.create_workspace('Test')
res_path = r"C:\Users\AbhilashKumar\Downloads\Result 1.mat"
a = workspace.upload_result(res_path ).wait()
a.delete()
```
@andrewwren-modelon I have a few questions on the API end - 
1. Is it possible to upload any file through this API, not just mat? What I observed that any file below 100 MB can be uploaded and the files are renamed to result.mat. 
2. Does the API POST/uploads/results hold the execution until the file is uploaded ? I wasn't able to slow it down so that the wait function can come into effect even with a larger file so was unsure of the behavior.

### CODE REVIEW

File: modelon/impact/client/operations.py [hunk 1]
```diff
@@ -23,9 +25,23 @@ class Status(Enum):
     DONE = "done"
 
 
-class Operation(ABC):
+@enum.unique
+class UploadStatus(Enum):
```
Comment (efredriksson-modelon) [inline modelon/impact/client/operations.py:29]: This one is actually a bit more generall than `upload` for example we have this status also for workspace export. So I suggest calling this status 'AsyncOperationStatus'.

File: modelon/impact/client/operations.py [hunk 2]
```diff
@@ -57,6 +73,64 @@ def name(self):
...
 
+    @abstractmethod
+    def wait(self, timeout):
+        """
+        Waits for the operation to finish.
+        """
+        pass
+
+
+class FileOperation(BaseOperation):
```
Comment (efredriksson-modelon) [inline modelon/impact/client/operations.py:84]: And then this one should be called `AsyncOperation`

File: tests/impact/client/fixtures.py [hunk 3]
```diff
@@ -233,6 +233,87 @@ def import_fmu(sem_ver_check, mock_server_base):
...
+
+@pytest.fixture
+def upload_result_meta(sem_ver_check, mock_server_base):
+    json = {
+        "id": "2f036b9fab6f45c788cc466da327cc78workspace",
+        "createdAt": 1588016253,
+        "name": "result_for_PID",
+        "description": "This is a result file for PID controller",
+        "workspaceId": "workspace",
+    }
```
Comment (efredriksson-modelon) [inline tests/impact/client/fixtures.py:298]: I think the API would return {'data': ...}, this was my point with one of the issues during release testing. Can you dubble check it?

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Good work 👍 Just a few minor comments.

File: modelon/impact/client/entities.py [hunk 4]
```diff
@@ -1996,6 +2053,43 @@ def keys(self):
...
+
+    @property
+    def metadata(self):
+        """External result metadata."""
+        upload_meta = self._workspace_sal.get_uploaded_result_meta(self._result_id)[
+            "data"
+        ]
+        id = upload_meta.get("id")
+        name = upload_meta.get("name")
+        created_at = upload_meta.get("createdAt")
```
Comment (efredriksson-modelon) [inline modelon/impact/client/entities.py:2084]: If we want to expose the date we should create a datetime object. Here is a stackoverflow on how to convert the isoformat string: https://stackoverflow.com/questions/127803/how-do-i-parse-an-iso-8601-formatted-date. So we would need to install the python-dateutil package to have it work with Python 3.6.

Or we could just remove it for now.

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Just one thing about exposing the date to the user.

File: modelon/impact/client/entities.py (hunk 4 above)
Comment (abhilash-kumar-nair) [reply modelon/impact/client/entities.py]: I have removed it for now. Thanks! 😄

**Pull request merged**